### PR TITLE
feat: Add Logs

### PR DESCRIPTION
**Pull request by arihanv**

### CODE REVIEW

File: app/[slug]/(base)/usage/[[...metric]]/layout.tsx
Comment (tedspare) [inline app/[slug]/(base)/usage/[[...metric]]/layout.tsx:33]: Should this code be uncommented?

File: lib/agents/engineer.ts
Comment (tedspare) [inline lib/agents/engineer.ts:32]: Have you tested the new model thoroughly? I tried it earlier and found weird results from it taking instructions too literally

File: lib/agents/engineer.ts
Comment (tedspare) [inline lib/agents/engineer.ts:54]: This is 🔥 , so clean

File: lib/agents/reviewer.ts
Comment (tedspare) [inline lib/agents/reviewer.ts:51]: Much cleaner, nice work

File: lib/components/dashboard/Metrics/LogsView.tsx
Comment (tedspare) [inline lib/components/dashboard/Metrics/LogsView.tsx:45]: What happens if `logs?.length == 0`?

File: lib/components/dashboard/Metrics/TableWrapper.tsx
Comment (tedspare) [inline lib/components/dashboard/Metrics/TableWrapper.tsx:80]: Commented code

File: lib/components/dashboard/Metrics/TableWrapper.tsx
Comment (tedspare) [inline lib/components/dashboard/Metrics/TableWrapper.tsx:94]: Commented code

File: lib/components/dashboard/Metrics/TableWrapper.tsx
Comment (tedspare) [inline lib/components/dashboard/Metrics/TableWrapper.tsx:129]: u kno the vibes

**Review by tedspare — COMMENTED** [review]
Big lift so far. A few nitpicks but nothing major if it all works.

Will wait for your merge request for the schema changes

File: lib/components/dashboard/Metrics/TableWrapper.tsx
Comment (arihanv) [reply lib/components/dashboard/Metrics/TableWrapper.tsx:80]: Commented since search can be added later, not supported atm since idk which column we actually would use search for.

File: lib/components/dashboard/Metrics/TableWrapper.tsx
Comment (arihanv) [reply lib/components/dashboard/Metrics/TableWrapper.tsx:94]: Same thing with prev comment

File: lib/components/dashboard/Metrics/TableWrapper.tsx
Comment (arihanv) [reply lib/components/dashboard/Metrics/TableWrapper.tsx:129]: Same thing with prev comment

File: lib/agents/engineer.ts
Comment (arihanv) [reply lib/agents/engineer.ts:32]: My tests with it for engineering and labeling went pretty well. Was super consistent for me. It should be the same model as gpt-4-1106

File: app/[slug]/(base)/usage/[[...metric]]/layout.tsx
Comment (arihanv) [reply app/[slug]/(base)/usage/[[...metric]]/layout.tsx:33]: Yeah, charts isn't supported now since the usage schema is very different

File: lib/components/dashboard/Metrics/LogsView.tsx
Comment (arihanv) [reply lib/components/dashboard/Metrics/LogsView.tsx:45]: Will add a state for that, good catch!

**Review by tedspare — APPROVED** [review]
LGTM if well-tested

**Pull request merged**
